File: ios/Classes/FlutterSmsPlugin.m
#import "FlutterSmsPlugin.h"
#import <send_message/send_message-Swift.h>

@implementation FlutterSmsPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  [SwiftFlutterSmsPlugin registerWithRegistrar:registrar];
}
@end

File: macos/Classes/FlutterSmsPlugin.swift
import Cocoa
import FlutterMacOS

public class FlutterSmsPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "flutter_sms", binaryMessenger: registrar.messenger)
    let instance = FlutterSmsPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("macOS " + ProcessInfo.processInfo.operatingSystemVersionString)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
